indico/modules/events/layout/controllers/layout.py

```diff
@@ -95,14 +96,24 @@ def _process(self):
 class RHLayoutLogoUpload(RHLayoutBase):
     def _process(self):
         f = request.files['file']
-        content = f.read()
+        try:
+            img = Image.open(f)
+        except IOError:
+            flash(_("You can't upload this file as a logo."))
+            return jsonify_data(content={})
+        if img.format.lower() not in {'jpg', 'jpeg', 'png', 'gif'}:
```

Comment:
I still it's better to figure out how you ended up with 'jpg' here and if you can't reproduce it remove it again...